docs/DesignDocs/Attestation_API_Proposal.md

```diff
@@ -0,0 +1,534 @@
+Proposal of OE SDK Attestation Public and Plugin API for V0.10 Release
+====
+
+# Background
+
+This document describes the proposed OE SDK attestation public and plugin API
+for V0.10 release, as an evolution of the experimental attestation API
+implemented in the
+[OE SDK V0.9 release](https://github.com/openenclave/openenclave/tree/v0.9.x).
+
+This proposal is based on these efforts:
+- Design document [Attestation V3 Update]:
+[Custom Attestation Data Formats for Open Enclave V3 Updates](https://github.com/openenclave/openenclave/blob/master/docs/DesignDocs/CustomAttestation_V3.md).
+- Discussion in [Issue #2729]: [Outstanding issues on attestation plugin design](https://github.com/openenclave/openenclave/issues/2729).
+- Design document [quote-ex Integration]:
+[Attestation: OE SDK Integration with Intel® SGX SDK quote-ex Library for Generation of Evidence in New Formats](https://github.com/openenclave/openenclave/blob/master/docs/DesignDocs/SGX_QuoteEx_Integration.md).
+- Design document [Design Notes]:
+[Notes on OE SDK Attestation API and Plugin Library Design](https://github.com/shnwc/openenclave/blob/master/docs/DesignDocs/NotesOnAttestationAPI.md).
+  - Note: this document is in the process of being up-streamed as OE SDK [PR #2801](https://github.com/openenclave/openenclave/pull/2801).
+
+# OE SDK Attestation Public API Proposal
+
+## Existing OE SDK V0.9 Public API
+
+The OE SDK V0.9 experimental public API for attestation is composed of
+functions for a number of use cases to be implemented by applications.
+The use cases fall into two categories: evidence generation and verification.
+
+Use cases for evidence generation:
+
+- Retrieve a single SGX attester plugin.
+  - Function `oe_attester_t* oe_sgx_plugin_attester(void)`,
```

Comment:
I'd recommend that all APIs have a verb.  So oe_sgx_get_plugin_attester() or oe_get_sgx_plugin_attester(), if this API is needed.

Reply:
The proposed new API will do.

Reply:
This is the v0.9 API